packages/maskbook/src/social-network-provider/twitter.com/ui/custom.ts

```diff
@@ -4,23 +4,26 @@ import { unstable_createMuiStrictModeTheme, ThemeProvider, makeStyles } from '@m
 import { useMaskbookTheme } from '../../../utils/theme'
 import type { SocialNetworkUICustomUI } from '../../../social-network/ui'
 import { useValueRef } from '../../../utils/hooks/useValueRef'
-import { composeAnchorSelector } from '../utils/selector'
+import { composeAnchorSelector, composeAnchorTextSelector } from '../utils/selector'
 import React from 'react'
-import { toRGB, getBackgroundColor, fromRGB, shade, isDark } from '../../../utils/theme-tools'
+import { toRGB, getBackgroundColor, fromRGB, shade, isDark, getForegroundColor } from '../../../utils/theme-tools'
 import { Appearance } from '../../../settings/settings'
 import produce, { setAutoFreeze } from 'immer'
 import type { InjectedDialogClassKey } from '../../../components/shared/InjectedDialog'
 import type { StyleRules } from '@material-ui/core'
 
 const primaryColorRef = new ValueRef(toRGB([29, 161, 242]))
+const primaryColorContrastColorRef = new ValueRef(toRGB([255, 255, 255]))
```

Comment:
Hmm, do you know mui provides a function in the theme to calculate contrast text color?

Reply:
But it returns the wrong result.

Reply:
For what color? Maybe we should report it to the upstream

Reply:
The default blue of Twitter. That function returns black as its contrast color.

Reply:
![](https://user-images.githubusercontent.com/5390719/98062366-d998fe00-1e88-11eb-9d85-015c0e74ed94.png)